3rdparty/p256-m/p256-m_driver_entrypoints.c

```diff
@@ -149,21 +229,23 @@ psa_status_t p256_transparent_sign_hash(
     (void) attributes;
     (void) alg;
 
-    psa_status_t status = PSA_ERROR_NOT_SUPPORTED;
-    if (key_buffer_size != 32 || signature_size != 64) {
-        return status;
+    /* Validate sizes, as p256-m expects fixed-size buffers */
+    if (key_buffer_size != PRIVKEY_SIZE) {
+        return PSA_ERROR_CORRUPTION_DETECTED;
```

Comment:
My reasoning is that (1) the key has been validated on import, and (2) driver dispatch should only call this driver for keys on the right curve. So, if this check fails, it means the core and/or dispatch is misbehaving.

We could almost skip the checks and just trust the core (this is a transparent driver, so there's no security barrier between the core and the driver) , but since this function wraps an API with "no explicit length because it's 32 all the time" into an interface with "explicit length parameter", I felt it was cleaner for the wrapper to assert that the length is as expected.

Reply:
I created a straw poll on slack to see what others think.

Reply:
I was going to ask a bit about this. I feel like `_CORRUPTION_DETECTED` is maybe too precise for a situation that may not be caused by corruption. This is a bit hand-wavey of me but I feel like we could get an incorrectly sized key for multiple reasons (a bug elsewhere we have yet to find or some part of the integration going wrong or indeed corruption) and so `_INVALID_ARGUMENT` could be a better suited catch-all error to report.

Reply:
Ok, I'll change those to `INVALID_ARGUMENT` then!

Reply:
Cool, ok thanks!

Reply:
Indeed, thinking about this a bit more I think `CORRUPTION_DETECTED` should be reserved for cases that are impossible, short of physical corruption during execution or a major bug in the code. We shouldn't use this for the kind of mistakes that can happen during configuration or integration, such as calling a driver on a key type that it doesn't support.

`CORRUPTION_DETECTED` should only catch bugs within code that's maintained by the same entity. The code path here involves the PSA core, a plug-in which is maintained in the same repository but in a separate directory under the heading “third-party”, possibly in the future a plug-in related to key parsing, a driver description file which may have been modified by a distributor or integrator with only vague knowledge of the various components involved, and a driver description generator/compiler script. With that many entities, if an error happens, we should convey “integration mistake”, not “memory corruption”.

Here either `NOT_SUPPORTED` (“I don't understand this, I guess it's a curve I don't support”) or `INVALID_ARGUMENT` (“I don't understand this, I have no idea what you mean”) seem appropriate.